File: Examples/python/can_pass.py
#!/usr/bin/python3

import sys
import time
import redpitaya_scpi as scpi

#rp_s = scpi.scpi(sys.argv[1])
rp_s = scpi.scpi("200.0.0.37")

rp_s.tx_txt('CAN:FPGA ON')
print("CAN:FPGA ON")
rp_s.check_error()

rp_s.tx_txt('CAN0:STOP')
print("CAN0:START")
rp_s.check_error()

rp_s.tx_txt('CAN1:STOP')
print("CAN1:STOP")
rp_s.check_error()

rp_s.tx_txt('CAN0:BITRate 200000')
print("CAN0:BitRate 200000")
rp_s.check_error()

rp_s.tx_txt('CAN1:BITR 200000')
print("CAN1:BitRate 200000")
rp_s.check_error()

rp_s.tx_txt('CAN0:MODE LOOPBACK,OFF')
print("CAN0:MODE LOOPBACK,OFF")
rp_s.check_error()

rp_s.tx_txt('CAN1:MODE LOOPBACK,OFF')
print("CAN1:MODE LOOPBACK,OFF")
rp_s.check_error()

rp_s.tx_txt('CAN0:START')
print("CAN0:START")
rp_s.check_error()

rp_s.tx_txt('CAN1:START')
print("CAN1:START")
rp_s.check_error()

rp_s.tx_txt('CAN0:OPEN')
print("CAN0:OPEN")
rp_s.check_error()

rp_s.tx_txt('CAN1:OPEN')
print("CAN1:OPEN")
rp_s.check_error()

rp_s.tx_txt('CAN0:Send123 1,2,3')
print("CAN0:SEND123 1,2,3")
rp_s.check_error()

rp_s.tx_txt('CAN0:Send321:E 1,2,3,4,5')
print("CAN0:Send321:E 1,2,3,4,5")
rp_s.check_error()

rp_s.tx_txt('CAN1:READ:TIMEOUT2000?')
print("CAN1:READ:TIMEOUT2000??",rp_s.rx_txt_check_error())

rp_s.tx_txt('CAN1:READ?')
print("CAN1:READ?",rp_s.rx_txt_check_error())

rp_s.tx_txt('CAN0:CLOSE')
print("CAN0:CLOSE")
rp_s.check_error()

rp_s.tx_txt('CAN1:CLOSE')
print("CAN1:CLOSE")
rp_s.check_error()

File: Examples/Matlab/can_pass.m
%% Define Red Pitaya as TCP/IP object

IP= '';           % Input IP of your Red Pitaya...
port = 5000;
tcpipObj=tcpip(IP, port);

%% Open connection with your Red Pitaya

fopen(tcpipObj);
tcpipObj.Terminator = 'CR/LF';
fprintf(tcpipObj,'CAN:FPGA ON');

fprintf(tcpipObj,'CAN0:STOP');             % stop can0 interface for configure

fprintf(tcpipObj,'CAN0:BITRate 200000');   % set bitrate for can0

fprintf(tcpipObj,'CAN0:MODE LOOPBACK,OFF'); % set loopback mode off

fprintf(tcpipObj,'CAN1:STOP');             % stop can1 interface for configure

fprintf(tcpipObj,'CAN1:BITRate 200000');   % set bitrate for can1

fprintf(tcpipObj,'CAN1:MODE LOOPBACK,OFF'); % set loopback mode  off

fprintf(tcpipObj,'CAN0:START');            % start can0 interface 

fprintf(tcpipObj,'CAN0:OPEN');             % open can0 socket 

fprintf(tcpipObj,'CAN1:START');            % start can1 interface 

fprintf(tcpipObj,'CAN1:OPEN');             % open can1 socket 

fprintf(tcpipObj,'CAN0:Send123 1,2,3');    % write to can0 3 bytes
fprintf('CAN0:Send123 1,2,3\n');

fprintf(tcpipObj,'CAN0:Send321:E 1,2,3,4,5');    % write to can0 3 bytes
fprintf('CAN0:Send321:E 1,2,3,4,5\n');

res = query(tcpipObj,'CAN1:Read:Timeout2000?'); % read frame from can1
fprintf('Read: %s', res);

res = query(tcpipObj,'CAN1:Read?');         % read frame from can1
fprintf('Read: %s', res);

fprintf(tcpipObj,'CAN0:CLOSE');            % close can0

fprintf(tcpipObj,'CAN1:CLOSE');            % close can1

%% Close connection with Red Pitaya

fclose(tcpipObj);